inspire_records/api/base.py

```diff
@@ -31,8 +31,45 @@
 from inspire_schemas.api import validate as schema_validate
 from invenio_pidstore.errors import PIDDoesNotExistError
 from invenio_records_files.api import Record
+from invenio_records.models import RecordMetadata
 from invenio_db import db
 
+from sqlalchemy import Text, or_, not_, cast, type_coerce
+from sqlalchemy.dialects.postgresql import JSONB
+
+class InspireQueryBuilderError(Exception):
+    pass
+
+class InspireQueryBuilder:
+
+    def __init__(self):
+        self._query = RecordMetadata.query 
+        self._filters = []
+
+    def not_deleted(self):
+        expression = or_(
+                        not_(
+                            type_coerce(RecordMetadata.json, JSONB).has_key('deleted')),
+                            not_(RecordMetadata.json['deleted'] == cast(True, JSONB)
+                        )
+                    )
```

Comment:
did `black` produce this? it has really funny indenting. At first sight it looks like the structure is `or_(not_(..., ...))`. If it was produced by `black`, I think it's worth a bug report.